test/e2e/custom_resource_definition.go

```diff
@@ -0,0 +1,198 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+type Noxu struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	SomeField  string `json:"someField"`
+	OtherField int    `json:"otherField"`
+}
+
+type NoxuList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" description:"standard list metadata; see http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata"`
+
+	Items []Noxu `json:"items"`
+}
+
+// This test is marked flaky pending namespace controller observing dynamic creation of new third party types.
+var _ = Describe("CustomResourceDefinition resources", func() {
+
+	f := framework.NewDefaultFramework("customresourcedefinition")
+
+	crd := &v1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "noxus.mygroup.example.com",
+		},
+		Spec: v1beta1.CustomResourceDefinitionSpec{
+			Group:   "mygroup.example.com",
+			Version: "v1alpha1",
+			Names: v1beta1.CustomResourceDefinitionNames{
+				Plural:   "noxus",
+				Singular: "noxu",
+				Kind:     "Noxu",
+				ListKind: "NoxuList",
+			},
+			Scope: v1beta1.ClusterScoped,
+		},
+	}
+
+	Context("Simple CustomResourceDefinition", func() {
+		It("creating/deleting custom resource definition objects works [Conformance]", func() {
+			config, err := framework.LoadConfig()
+			if err != nil {
+				framework.Failf("failed to load config: %v", err)
+			}
+
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			if err != nil {
+				framework.Failf("failed to initialize apiExtensionClient: %v", err)
+			}
+
+			defer func() {
+				if err := apiExtensionClient.Apiextensions().CustomResourceDefinitions().Delete(crd.Name, nil); err != nil {
+					framework.Failf("failed to delete custom resource definitions resource: %v", err)
+				}
+			}()
+
+			if _, err := apiExtensionClient.Apiextensions().CustomResourceDefinitions().Create(crd); err != nil {
+				framework.Failf("failed to create custom resource definitions resource: %v", err)
+			}
+
+			wait.Poll(time.Second*30, time.Minute*5, func() (bool, error) {
```

Comment:
instead of this, create a helper that takes a CRD, creates the CRD, waits for the status to show available, and returns a defer function you can run to clean it up.  After you call that function, you have no need to wait here.